src/app/services/actions/autocomplete-action-creators.ts

```diff
@@ -0,0 +1,91 @@
+import { IAction } from '../../../types/action';
+import { IRequestOptions } from '../../../types/request';
+import { AUTOCOMPLETE_FETCH_ERROR, AUTOCOMPLETE_FETCH_PENDING, AUTOCOMPLETE_FETCH_SUCCESS } from '../redux-constants';
+
+export function fetchAutocompleteSuccess(response: object): IAction {
+  return {
+    type: AUTOCOMPLETE_FETCH_SUCCESS,
+    response,
+  };
+}
+
+export function fetchAutocompleteError(response: object): IAction {
+  return {
+    type: AUTOCOMPLETE_FETCH_ERROR,
+    response,
+  };
+}
+
+export function fetchAutocompletePending(): any {
+  return {
+    type: AUTOCOMPLETE_FETCH_PENDING
+  };
+}
+
+export function fetchAutoCompleteOptions(url: string): Function {
+  return async (dispatch: Function, getState: Function) => {
+    const devxApi = getState().devxApi;
+    const sampleQuery = getState().sampleQuery;
+    const headers = {
+      'Content-Type': 'application/json',
+    };
+
+    let permissionsUrl = `${devxApi}/openapi`;
+    permissionsUrl = `${permissionsUrl}?url=/${url}&style=geautocomplete&format=json`;
```

Comment:
It is misleading to refer to this as the `permissionUrl` especially because we have a permissions tab.

Reply:
makes sense. Fixed this [here](https://github.com/microsoftgraph/microsoft-graph-explorer-v4/pull/489/commits/8b38678b5e830b3bbf96df0d4191f6ab7f84e104)